apps/shared/test/validation/jsonschema_validator_test.exs

```diff
@@ -96,6 +96,64 @@ defmodule Shared.Validation.JSONSchemaValidatorTest do
 
       assert nil == validate(name_jsonschema(), url)
     end
+
+    test "supports version up to 7" do
+      base_schema = %{"type" => "object", "properties" => %{"foo" => %{"type" => "string"}}}
+      v7_schema = Map.put(base_schema, "$schema", "http://json-schema.org/draft-07/schema")
+
+      assert ExJsonSchema.Schema.resolve(v7_schema)
+
+      # 2019-09 was previously known as version 8
+      assert_raise ExJsonSchema.Schema.UnsupportedSchemaVersionError, fn ->
+        base_schema
+        |> Map.put("$schema", "https://json-schema.org/draft/2019-09/schema")
+        |> ExJsonSchema.Schema.resolve()
+      end
+    end
+
+    test "can validate dependencies" do
+      schema = %ExJsonSchema.Schema.Root{
+        schema: %{
+          "properties" => %{"name" => %{"type" => "string"}, "age" => %{"type" => "integer"}},
+          "type" => "object",
+          "dependencies" => %{"name" => ["age"], "age" => ["name"]}
+        },
+        version: 7
+      }
+
+      assert %{
+               "errors" => ["#: Property name depends on property age to be present but it was not."],
+               "errors_count" => 1,
+               "has_errors" => true,
+               "validator" => Shared.Validation.JSONSchemaValidator
+             } == validate(schema, %{"name" => "Bob"})
+
+      assert %{
+               "errors" => ["#: Property age depends on property name to be present but it was not."],
+               "errors_count" => 1,
+               "has_errors" => true,
+               "validator" => Shared.Validation.JSONSchemaValidator
+             } == validate(schema, %{"age" => 42})
+
+      # Our Elixir validator does not recognise `dependentRequired` yet,
+      # even if it exists in the JSONSchema spec
+      # https://json-schema.org/understanding-json-schema/reference/conditionals.html#id4
+      #
+      # See https://github.com/etalab/transport-site/issues/2347
+      dependent_required_schema = %ExJsonSchema.Schema.Root{
+        schema:
+          Enum.into(schema.schema, %{}, fn
+            {"dependencies", value} -> {"dependentRequired", value}
+            pair -> pair
+          end),
```

Comment:
Si vous avez plus élégant pour renommer une seule clé d'une `Map` je prends

Reply:
Je te propose quelque chose qui évite d'itérer:

```elixir
    map = %{
      "dependencies" => "1-2-3"
    }

    # lèvera une erreur si la clé est absente
    {value, map} = Map.pop!(map, "dependencies")
    map = Map.put(map, "dependentRequired", value)
```